### PR TITLE
refactor: extract inbound transporter from agent constructor

### PR DESCRIPTION
**Pull request by jakubkoci**
It aligns API with the setting of the outbound transporter. It's also better for the cases when a developer wants to pass messages into the agent by themself.

BREAKING CHANGE

Signed-off-by: Jakub Koci <jakub.koci@gmail.com>

### CODE REVIEW

File: src/agent/Agent.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ export class Agent {
   protected messageReceiver: MessageReceiver
   protected dispatcher: Dispatcher
   protected messageSender: MessageSender
+  public inboundTransporter?: InboundTransporter
```
Comment (TimoGlastra) [inline src/agent/Agent.ts:43]:
```suggestion
  public readonly inboundTransporter?: InboundTransporter
```

File: src/agent/Agent.ts [hunk 2]
```diff
@@ -147,7 +146,9 @@ export class Agent {
       })
     }
 
-    return this.inboundTransporter.start(this)
+    if (this.inboundTransporter) {
+      await this.inboundTransporter.start(this)
+    }
```
Comment (TimoGlastra) [inline src/agent/Agent.ts:151]: Maybe we should log a warning for both the inbound/outbound transporter if they are not set at initialization yet. Just for information

**Review by TimoGlastra — APPROVED** [review]

**Comment by TimoGlastra** [comment]
Please merge my other PR #244 first. Otherwise I need to do quite some manual work again

File: src/agent/Agent.ts (hunk 2 above)
Comment (jakubkoci) [reply src/agent/Agent.ts:151]: Hmm, actually I can imagine it's not a problem not having an inbound transported set. But at least the info could be good. For example, I run AFJ as a mediator with Nest.js framework and it's quite convenient I don't have to set inbound transport but I handle it by myself.

**Pull request merged**